plugins/autocomplete/plugin.js

```diff
@@ -0,0 +1,1208 @@
+			this.element.setHtml( '' );
+			this.element.append( itemsFragment );
+		},
+
+		/**
+		 * Attaches the view's listeners to the DOM elements.
+		 */
+		attach: function() {
+			this.element.on( 'click', function( evt ) {
+				var target = evt.data.getTarget(),
+					itemElement = target.getAscendant( this.isItemElement, true );
+
+				if ( itemElement ) {
+					this.fire( 'click-item', itemElement.data( 'id' ) );
+				}
+			}, this );
+		},
+
+		/**
+		 * Closes the panel.
+		 */
+		close: function() {
+			this.element.removeClass( 'cke_autocomplete_opened' );
+		},
+
+		/**
+		 * Creates and returns the view's main element.
+		 *
+		 * @returns {CKEDITOR.dom.element}
+		 */
+		createElement: function() {
+			var el = new CKEDITOR.dom.element( 'ul', this.document );
+
+			el.addClass( 'cke_autocomplete_panel' );
+			// Below float panels and context menu, but above maximized editor (-5).
+			el.setStyle( 'z-index', this.editor.config.baseFloatZIndex - 3 );
+
+			return el;
+		},
+
+		/**
+		 * Creates the item element based on the {@link #itemTemplate}.
+		 *
+		 * @param {CKEDITOR.plugins.autocomplete.model.item} item The item for which an element will be created.
+		 * @returns {CKEDITOR.dom.element}
+		 */
+		createItem: function( item ) {
+			return CKEDITOR.dom.element.createFromHtml( this.itemTemplate.output( item ), this.document );
+		},
+
+		/**
+		 * Returns the caret position relative to the panel's offset parent.
+		 * The value returned by this function is passed to the {@link #setPosition} method
+		 * by the {@link #updatePosition} method.
+		 *
+		 * @returns {Object} Represents the position of the caret.
+		 * @returns {Number} rect.left
+		 * @returns {Number} rect.top
+		 * @returns {Number} rect.bottom
+		 */
+		getCaretRect: function() {
+			var caretClientRect = this.editor.getSelection().getCaretPosition(),
+				offset,
+				editable = this.editor.editable();
+
+			if ( editable.isInline() ) {
+				offset = CKEDITOR.document.getWindow().getScrollPosition();
+			} else {
+				offset = editable.getParent().getDocumentPosition( CKEDITOR.document );
+			}
+
+			// Consider that offset host might be repositioned on its own.
+			// Similar to #1048. See https://github.com/ckeditor/ckeditor-dev/pull/1732#discussion_r182790235.
+			var hostElement = CKEDITOR.document.getBody();
+			if ( hostElement.getComputedStyle( 'position' ) === 'static' ) {
+				hostElement = hostElement.getParent();
+			}
+
+			var offsetCorrection = hostElement.getDocumentPosition();
+
+			offset.x -= offsetCorrection.x;
+			offset.y -= offsetCorrection.y;
+
+			return {
+				top: ( caretClientRect.top + offset.y ),
+				bottom: ( caretClientRect.top + caretClientRect.height + offset.y ),
+				left: ( caretClientRect.left + offset.x )
+			};
+		},
+
+		/**
+		 * Gets the item element by the item ID.
+		 *
+		 * @param {Number/String} itemId
+		 * @returns {CKEDITOR.dom.element} The item element.
+		 */
+		getItemById: function( itemId ) {
```

Comment:
Method doesn't handle situation when `itemId` contains specific characters. 
e.g.
Try to use as id this string: `:Antigua_&amp;_Barbuda:`